src/modules/common/helpers/mappings.ts

```diff
@@ -0,0 +1,73 @@
+import { IHomeCarouselData } from "../../home/types/home-types";
+import useMangaStore from "../stores/store";
+import { ITrendingMangaApp } from "../types/app-types";
+
+export const getViewAllCategoryHeading = (category: string) => {
+    switch (category) {
+        case "trending":
+            return "Trending Mangas"
+            break;
+        case "new":
+            return "Latest Mangas"
+            break;
+        default:
+            break;
+    }
+}
+
+export const getCarouselDetails = (category: string): IHomeCarouselData => {
+    switch (category) {
+        case 'new':
+            {
+                return { heading: "Fresh off The Pen", queryKey: "fetchNewMangas" }
+            }
+            break;
+        case 'rank':
+            {
+                return { heading: "Chart-Topping Manga Marvels", queryKey: "fetchTopRankedMangas" }
+            }
+            break;
+        case 'popular':
+            {
+                return { heading: "Tales that Hooked the Hoard", queryKey: "fetchMostFollowedMangas" }
+            }
+            break;
+
+        default:
+            return { heading: "", queryKey: "" }
+            break;
+    }
+}
+
+export const getHomeCarouselState = (category: string) => {
+    switch (category) {
+        case 'new':
+            return useMangaStore.getState().newMangas
+            break;
+        case 'rank':
+            return useMangaStore.getState().topRankedMangas
+            break;
+        case 'popular':
+            return useMangaStore.getState().popularMangas
+            break;
+        default:
+            return []
+            break;
+    }
+}
+
+export const setHomeCarouselState = (category: string, newState: ITrendingMangaApp[]) => {
+    switch (category) {
+        case 'new':
+            useMangaStore.setState({ newMangas: newState })
+            break;
+        case 'rank':
+            useMangaStore.setState({ topRankedMangas: newState })
+            break;
+        case 'popular':
+            useMangaStore.setState({ popularMangas: newState })
+            break;
+        default:
+            break;
```

Comment:
Maybe I am not fully understanding the code here, always make sure that the default case is handled accordingly. However, if it doesn't matter for **any** edge cases, then it should be fine.

Reply:
I left it blank on purpose. If the passed in category prop is not one of those options, the state should not be set. And since I will be hard-coding that prop, it will always be one of the valid options

Reply:
I understand that and there's no problem with it. In terms of the last sentence, we should write every component with the assumption that we don't fully know that the prop will always come back with the values we expect. Since in real projects other people will probably pass in the props from different components we should always cater for every case. Having said that, the code here does look good and the logic makes sense.👍